### PR TITLE
implement 1980:[Feature][Java Service] Update Service Pattern APIs to Support Level 2 Unit

### PR DESCRIPTION
**Pull request by huihuiw01**

### CODE REVIEW

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java [hunk 1]
```diff
@@ -207,21 +207,24 @@ private Map<String, Object> getCategories(List<String> categoryList, Map<String,
 		Map<String, Object> resultMap = new LinkedHashMap<>();
 		Map<String, Object> categoriesMap = (Map<String, Object>) patternMap.get(ConstantsKeys.CATEGORIES);
 		Map<String, Object> supplementMap = (Map<String, Object>) categoriesMap.get(ConstantsKeys.SUPPLEMENT);
-		Map<String, Object> suppMap = new HashMap<>();
-		for (String cat : categoryList) {
-			suppMap.put(cat, supplementMap.get(cat));
-		}
 
-		if (categoryList.contains(ConstantsKeys.CURRENCIES) && !categoryList.contains(ConstantsKeys.NUMBERS)) {
+		if (categoryList.contains(ConstantsKeys.CURRENCIES) || categoryList.contains(ConstantsKeys.MEASUREMENTS) || categoryList.contains(ConstantsKeys.DATE_FIELDS) && !categoryList.contains(ConstantsKeys.PLURALS)) {
```
Comment (Xiaochao8) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java:211]: Why does currency need plural?

Comment (huihuiw01) [reply g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java]: currency is a special kind of measurement, it also has single and plural forms, please refer to the cldr data:
https://github.com/unicode-cldr/cldr-numbers-full/blob/32.0.0/main/br/currencies.json

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java [hunk 2]
```diff
@@ -207,21 +207,29 @@ private Map<String, Object> getCategories(List<String> categoryList, Map<String,
...
 				resultMap.put(cat, categoriesMap.get(cat));
 			}
 		}
+
+		Map<String, Object> suppMap = new HashMap<>();
+		for (String cat : categoryList) {
+			if (!CommonUtil.isEmpty(supplementMap.get(cat))) {
+				suppMap.put(cat, supplementMap.get(cat));
+			}
+		}
```
Comment (linr211) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java:232]: seems you can combine the two 'for' loops as one.

File: g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java [hunk 3]
```diff
@@ -207,21 +207,29 @@ private Map<String, Object> getCategories(List<String> categoryList, Map<String,
...
 
 		for (String cat : categoryList) {
 			if (!CommonUtil.isEmpty(categoriesMap.get(cat))) {
 				resultMap.put(cat, categoriesMap.get(cat));
 			}
 		}
+
+		Map<String, Object> suppMap = new HashMap<>();
+		for (String cat : categoryList) {
+			if (!CommonUtil.isEmpty(supplementMap.get(cat))) {
```
Comment (tigershi) [inline g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java:229]: Can you move the if (!CommonUtil.isEmpty(supplementMap.get(cat))) {suppMap.put(cat, supplementMap.get(cat));
			} to line 224

**Review by tigershi — APPROVED** [review]

**Review by linr211 — APPROVED** [review]

**Pull request merged**